protocol/xmrmaker/instance.go

```diff
@@ -198,6 +188,48 @@ func (inst *Instance) createOngoingSwap(s *swap.Info) error {
 	return nil
 }
 
+func (inst *Instance) completeSwap(s *swap.Info, skA *mcrypto.PrivateSpendKey) error {
+	// calc counterparty's swap private view key
+	// TODO: if the counterparty sends a non-standard public view key, does this work?
```

Comment:
It won't work. The names make it sound like you only need the spend key to spend, but in order to move funds around, you need both. I'm not sure on the technical reasons, but I think the wallet doesn't even know which transaction outputs belong to it without having the view key set correctly.

Reply:
yep I'm going to fix this in #304